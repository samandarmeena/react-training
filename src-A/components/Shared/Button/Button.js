import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.module.scss';

/**
 * Button component
 * It renders button
 * @param {string} buttonType
 * @param {string} buttonName
 * @param {string} buttonId
 * @param {string} buttonTitle
 * @param {string} buttonStyle
 * @param {string} buttonSize
 * @param {string} addOnClass
 * @param {string} buttonToolTip
 * @param {fn} clickEvent
 * @return Element
 */
const Button = props => {
  const {
    buttonType,
    buttonName,
    buttonId,
    buttonTitle,
    buttonStyle,
    buttonSize,
    addOnClass,
    buttonToolTip,
    clickEvent
  } = props;
  return (
    <button
      type={buttonType}
      name={buttonName}
      id={buttonId}
      className={`${styles.button} ${styles[buttonStyle]} ${styles[buttonSize]} ${addOnClass}`}
      onClick={() => clickEvent()}
      title={buttonToolTip}
    >
      {buttonTitle}
    </button>
  );
};

// proptype validations for props
Button.propTypes = {
  buttonType: PropTypes.string,
  buttonName: PropTypes.string.isRequired,
  buttonId: PropTypes.string.isRequired,
  buttonTitle: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
  buttonStyle: PropTypes.string,
  buttonSize: PropTypes.string,
  addOnClass: PropTypes.string,
  buttonToolTip: PropTypes.string.isRequired,
  clickEvent: PropTypes.func
};

// default values for props
Button.defaultProps = {
  buttonType: 'button',
  buttonStyle: 'buttonPrimary',
  buttonSize: 'buttonMedium',
  addOnClass: ''
};

export default Button;
