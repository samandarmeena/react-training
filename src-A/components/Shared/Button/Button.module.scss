.button {
  border: none;
  cursor: pointer;
  margin-left: 3px;
  margin-right: 3px;
}
.buttonInfo {
  background-color: #138496;
  color: #fff;
}
.buttonInfo:hover {
  color: #fff;
  background-color: #007183;
  border-color: #007183;
}
.buttonPrimary {
  background-color: #0069d9;
  color: #fff;
}
.buttonPrimary:hover {
  background-color: #0052c2;
}
.buttonSuccess {
  background-color: #40c2a6;
  color: #fff;
}
.buttonWarning {
  background-color: #e0a800;
  color: #fff;
}
.buttonDanger {
  background-color: #c82333;
  color: #fff;
}
.buttonDanger:hover {
  background-color: #ab0616;
}
.buttonLarge {
  font-size: 17px;
  border-radius: 5px;
  padding: 13px;
}
.buttonMedium {
  font-size: 15px;
  border-radius: 5px;
  padding: 10px;
}
.buttonSmall {
  font-size: 12px;
  border-radius: 5px;
  padding: 5px;
}
