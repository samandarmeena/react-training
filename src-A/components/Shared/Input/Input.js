import React from 'react';
import PropTypes from 'prop-types';
import styles from './Input.module.scss';

/**
 * Input component
 * It renders input
 * @param {string} inputType
 * @param {string} inputName
 * @param {string} inputId
 * @param {string} inputValue
 * @param {string} inputPlaceholder
 * @return Element
 */
const Input = (props) => {
  const {
    inputType,
    inputName,
    inputId,
    inputValue,
    inputPlaceholder,
    changeEvent
  } = props;
  return (
    <input
      type={inputType}
      className={styles.inputBox}
      id={inputId}
      name={inputName}
      placeholder={inputPlaceholder}
      value={inputValue}
      onChange={changeEvent}
    />
  );
};

// proptype validations for props
Input.propTypes = {
  inputType: PropTypes.string,
  inputName: PropTypes.string.isRequired,
  inputId: PropTypes.string.isRequired,
  inputValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  inputPlaceholder: PropTypes.string.isRequired,
  changeEvent: PropTypes.func.isRequired
};

// default values for props
Input.defaultProps = {
  inputType: 'text'
};

export default Input;
