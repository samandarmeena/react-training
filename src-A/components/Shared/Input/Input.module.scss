.inputBox {
  width: 35%;
  border-radius: 5px;
  border: none;
  margin-right: 10px;
  padding: 10px;
  font-size: 15px;
}
