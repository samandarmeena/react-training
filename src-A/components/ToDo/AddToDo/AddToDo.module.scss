.addToDo {
  font-size: 16px;
}
.title {
  font-size: 25px;
  margin-bottom: 10px;
}
.message {
  font-size: 15px;
  font-size: 17px;
  color: #4d4d4d;
  margin-bottom: 15px;
}
.welcomeBox {
  background: #ccc;
  padding: 30px 15px 40px 15px;
  border: 1px solid #b5b5b5;
  text-align: center;
  border-radius: 5px;
}
