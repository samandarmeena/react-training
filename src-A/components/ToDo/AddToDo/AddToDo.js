import React from 'react';
import PropTypes from 'prop-types';
import Input from '../../Shared/Input';
import Button from '../../Shared/Button';
import styles from './AddToDo.module.scss';

/**
 * AddToDo component
 * renders AddToDo container
 * @param {string} todoTitle
 * @param {func} handleChange
 * @param {func} onSave
 * @return Element
 */
const AddToDo = (props) => {
  const { todoTitle, handleChange, onSave } = props;
  return (
    <div className={styles.welcomeBox}>
      <div className={styles.title}>Welcome!</div>
      <div className={styles.message}>
        To get started, add some items to your list.
      </div>
      <div className={styles.addToDo}>
        <Input
          inputType="text"
          inputName="todoTitle"
          inputId="todoTitle"
          inputPlaceholder="Enter here"
          inputValue={todoTitle}
          changeEvent={handleChange}
        />
        <Button
          buttonName="saveToDo"
          buttonId="saveToDo"
          buttonTitle="Add New"
          clickEvent={onSave}
          buttonToolTip="Add ToDo"
          addOnClass="saveToDoBtn"
        />
      </div>
    </div>
  );
};

// proptype validations for props
AddToDo.propTypes = {
  todoTitle: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired
};

export default AddToDo;
