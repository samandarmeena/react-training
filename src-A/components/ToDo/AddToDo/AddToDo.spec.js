import React from 'react';
import { mount } from 'enzyme';
import AddToDo from './AddToDo';
import Input from '../../Shared/Input';
import Button from '../../Shared/Button';

describe('AddToDo component', () => {
  let wrapper = '';
  let inputProps = {};
  let mockOnChange = jest.fn();
  let mockOnSave = jest.fn();
  beforeAll(() => {
    inputProps = {
      todoTitle: 'Test todo title 1',
      handleChange: mockOnChange,
      onSave: mockOnSave
    };
    wrapper = mount(<AddToDo {...inputProps} />);
  });

  it('AddToDo component rendered without an error', () => {
    expect(wrapper).toBeTruthy();
  });

  it('Check length of the children', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('Check children component rendered successfully', () => {
    expect(wrapper.find(Input).length).toBe(1);
    expect(wrapper.find(Button).length).toBe(1);
  });

  it('Should call save function one time', () => {
    wrapper
      .find(Input)
      .props()
      .changeEvent();
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  it('Should call save function one time', () => {
    wrapper
      .find(Button)
      .props()
      .clickEvent();
    expect(mockOnSave).toHaveBeenCalledTimes(1);
  });

  it('Snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
