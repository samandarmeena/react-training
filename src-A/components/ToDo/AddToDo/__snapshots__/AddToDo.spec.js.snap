// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToDo component Snapshot testing 1`] = `
<AddToDo
  handleChange={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onSave={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  todoTitle="Test todo title 1"
>
  <div
    className="welcomeBox"
  >
    <div
      className="title"
    >
      Welcome!
    </div>
    <div
      className="message"
    >
      To get started, add some items to your list.
    </div>
    <div
      className="addToDo"
    >
      <Input
        changeEvent={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        inputId="todoTitle"
        inputName="todoTitle"
        inputPlaceholder="Enter here"
        inputType="text"
        inputValue="Test todo title 1"
      >
        <input
          className="inputBox"
          id="todoTitle"
          name="todoTitle"
          onChange={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          placeholder="Enter here"
          type="text"
          value="Test todo title 1"
        />
      </Input>
      <Button
        addOnClass="saveToDoBtn"
        buttonId="saveToDo"
        buttonName="saveToDo"
        buttonSize="buttonMedium"
        buttonStyle="buttonPrimary"
        buttonTitle="Add New"
        buttonToolTip="Add ToDo"
        buttonType="button"
        clickEvent={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      >
        <button
          className="button buttonPrimary buttonMedium saveToDoBtn"
          id="saveToDo"
          name="saveToDo"
          onClick={[Function]}
          title="Add ToDo"
          type="button"
        >
          Add New
        </button>
      </Button>
    </div>
  </div>
</AddToDo>
`;
